### PR TITLE
[Intel MKL] Enabling BFloat16 NN ops using MKLDNN API - Round 1

### PR DESCRIPTION
**Pull request by nhasabni**
This PR enables support for BFloat16 type in all the pooling ops using MKLDNN APIs.

### CODE REVIEW

**Comment by reedwm** [comment]
I am not familiar with this code, so I'll leave the review to @penpornk

File: tensorflow/core/kernels/mkl_pooling_ops_common.cc [hunk 1]
```diff
@@ -217,6 +218,7 @@ void MklPoolingBwdPrimitive<T>::Execute(const T* diff_dst_data,
 }
 
 template class MklPoolingBwdPrimitive<float>;
+template class MklPoolingBwdPrimitive<bfloat16>;
 
 #endif
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_pooling_ops_common.cc]:
```suggestion
#endif  // !INTEL_MKL_MKL_ONLY
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm sorry for my delay! One change (unrelated to the PR), please.

File: tensorflow/core/kernels/mkl_pooling_ops_common.cc (hunk 1 above)
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_pooling_ops_common.cc]: thanks for review @penpornk. Pls take a look at my change.

**Review by penpornk — APPROVED** [review]
Thank you again for contributing! :)

**Pull request merged**